### PR TITLE
 :sparkles: Add selectablefield marker

### PR DESCRIPTION
**Pull request by everesio**
Add support for selectable fields for custom resources https://github.com/kubernetes-sigs/controller-tools/issues/1039

### CODE REVIEW

File: pkg/crd/markers/crd.go [hunk 1]
```diff
@@ -388,3 +391,32 @@ func (s Metadata) ApplyToCRD(crd *apiext.CustomResourceDefinition, _ string) err
...
+	var selectableFields *[]apiext.SelectableField
+	for i := range crd.Versions {
+		ver := &crd.Versions[i]
+		if ver.Name != version {
+			continue
+		}
+		selectableFields = &ver.SelectableFields
+		break
+	}
+	if selectableFields == nil {
```
Comment (sbueringer) [inline pkg/crd/markers/crd.go:413]: Not sure I get in which cases we would hit this case

File: pkg/crd/markers/crd.go [hunk 2]
```diff
@@ -55,6 +55,9 @@ var CRDMarkers = []*definitionWithHelp{
 
 	must(markers.MakeDefinition("kubebuilder:metadata", markers.DescribesType, Metadata{})).
 		WithHelp(Metadata{}.Help()),
+
+	must(markers.MakeDefinition("kubebuilder:selectablefield", markers.DescribesType, SelectableField{})).
```
Comment (sbueringer) [inline pkg/crd/markers/crd.go:59]: Let's add a few of these to https://github.com/kubernetes-sigs/controller-tools/blob/7ef21aae5452047af82f020774639c94aa659eca/pkg/crd/testdata/cronjob_types.go#L636

IIRC they then also have to be added here: https://github.com/kubernetes-sigs/controller-tools/blob/7ef21aae5452047af82f020774639c94aa659eca/pkg/crd/testdata/testdata.kubebuilder.io_cronjobs.yaml

(but you can just check the failing test after the marker was added to the type)

**Comment by sbueringer** [comment]
@everesio Do you have time to address the findings above?

**Comment by everesio** [comment]
@sbueringer Sorry, I thought the remarks were directed at the other reviewers.
I can address the findings. Let's clarify the issues:
1. Do you want to have some test which will the "if selectableFields == nil" check ? If similar test exists for printcolumn i can do it analogically.
2. I will add a new marker to the testdata.kubebuilder.io_cronjobs.yaml and check the parser_integration_test.go

**Comment by everesio** [comment]
I am just wondering if there is a regression, as the generated specification creates more differences than expected. This issue also occurs on the main branch without my changes.

The binary was built with `make release-binary` and later in the `pkg/crd/testdata`   the command `../../../out/controller-gen crd paths=. output:dir=.` was invoked


See result https://github.com/everesio/controller-tools/commit/fa0ae906c2fc6f89db426af0480bfe2dc5be9d66

Generated output with warnings:
```
➜  testdata git:(selectable-field-test) ../../../out/controller-gen crd paths=. output:dir=.
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:273:23: found float, the usage of which is highly discouraged, as support for them varies across languages. Please consider serializing your float as string instead. If you are really sure you want to use them, re-run with crd:allowDangerousTypes=true
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:273:2: must apply multipleof to a numeric value, found 
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:273:2: must apply minimum to a numeric value, found 
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:273:2: must apply maximum to a numeric value, found 
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:278:25: found float, the usage of which is highly discouraged, as support for them varies across languages. Please consider serializing your float as string instead. If you are really sure you want to use them, re-run with crd:allowDangerousTypes=true
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:278:2: must apply minimum to a numeric value, found 
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:278:2: must apply maximum to a numeric value, found 
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:278:2: must apply multipleof to a numeric value, found 
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:291:2: encountered struct field "unexportedField" without JSON tag in type "CronJobSpec"
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:423:2: encountered struct field "bar" without JSON tag in type "unexportedStruct"
/home/michal/go/src/sigs.k8s.io/controller-tools/pkg/crd/testdata/cronjob_types.go:431:2: encountered struct field "qux" without JSON tag in type "ExportedStruct"
Error: not all generators ran successfully
```

**Comment by everesio** [comment]
Test added. I've modified cronjob_types.go and testdata.kubebuilder.io_cronjobs.yaml

**Comment by sbueringer** [comment]
Thank you!!

/lgtm
/approve

**Pull request merged**